hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java

```diff
@@ -324,6 +332,32 @@ public StateView(
 		}
 	}
 
+	public Optional<TokenNftInfo> infoForNft(NftID nftID) {
+		try {
+			var exists = uniqueTokenStore.nftExists(nftID);
+			if (!exists) {
+				return Optional.empty();
+			}
+			var uniqueToken = uniqueTokenStore.get(nftID);
+
+			var info = TokenNftInfo.newBuilder()
+					.setNftID(nftID)
+					.setAccountID(uniqueToken.getOwner().toGrpcAccountId())
+					.setCreationTime(Timestamp.newBuilder()
+							.setSeconds(uniqueToken.getCreationTime().getSeconds())
+							.setNanos(uniqueToken.getCreationTime().getNanos()))
+					.setMetadata(ByteString.copyFromUtf8(uniqueToken.getMemo()));
```

Comment:
This should be updated once we update the `metadata` property as-well (@daniel-danev-limechain)